addons/mrp/tests/test_bom.py

```diff
@@ -1961,3 +1961,75 @@ def test_component_when_bom_change(self):
         # Check that component 1 is set again and component 2 is removed
         self.assertEqual(mo_form.move_raw_ids._records[0]['product_id'], self.product_1.id)
         self.assertEqual(len(mo_form.move_raw_ids._records), 1)
+
+    def test_update_operations(self):
+        """Update the operations in BoM which reflects the changes in Manufacturing Order"""
+
+        test_bom = self.env['mrp.bom'].create({
+            'product_tmpl_id': self.product_2.product_tmpl_id.id,
+            'product_uom_id': self.product_2.uom_id.id,
+            'product_qty': 1.0,
+            'type': 'normal',
+            'bom_line_ids': [
+                (0, 0, {'product_id': self.product_7_1.id, 'product_qty': 1.0}),
+            ],
+            'operation_ids': [
+                Command.create({
+                    'name': 'Cutting',
+                    'workcenter_id': self.workcenter_1.id,
+                    'sequence': 1
+                })
+            ],
+        })
+        mo_form = Form(self.env['mrp.production'].with_user(self.user_mrp_user))
+        mo_form.product_id = self.product_7_1
+        mo_form.product_qty = 1.0
+        mo_form.bom_id = test_bom
+        mo = mo_form.save()
+        mo.action_confirm()
+
+        test_bom.operation_ids.update({
+            'name': 'Painting',
+            'workcenter_id': self.workcenter_2.id
+        })
```

Comment:
I'm not sure when you should ever use `.update()`, `.write()` is more common though, can you use that instead?

Your test also made me realize that we should add `_set_outdated_bom_in_productions` to an override of the `create` and `write` of `mrp.routing.workcenter`, otherwise the "Update BoM" button won't show when the operation is directly created/edited via Configuration > Operations. (To be confirmed with the PO that we should fix this use case). Note that you will probably need to remove `operation_ids` from the bom write's `relevant_fields = ['bom_line_ids', 'byproduct_ids', 'product_tmpl_id', 'product_id', 'product_qty', 'operation_ids']` so that we can try avoiding calling it twice when operations are edited via the bom.

Also, after the write you should add another assert that `is_outdated_bom = True` once you make the above fix as well. If we don't apply the above fix, then you should write on the bom directly and then add the assert